pkg/util/tls/generate_cert.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2025 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tls
+
+import (
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"time"
+)
+
+// Adapted from test certificate generation in src/crypto/tls.
```

Comment:
Should we move this to the test folder if it's only used for testing purpose? or we may expect to use these in other features as well in the future?

Reply:
I am not sure which test folder you have in mind. This is used for both unit and e2e tests.
However, I have renamed the package to `pkg/util/tlstest`, which matches another similar package that we have (`pkg/agent/util/nettest`), also used for testing.